BuildConfigGen/VersionParser.cs

```diff
@@ -1,40 +1,130 @@
-﻿internal static class VersionParser
+﻿public static class VersionParser
 {
+    /// <summary>
+    /// Splits the full Semver 2.0 string into individual string components: version, pre-release version, and build version.
+    /// The actual values are not validated for Semver 2.0 version compliance.
+    /// </summary>
+    /// <param name="fullVersion">The complete version string, e.g. 1.2.3-test+node</param>
+    /// <param name="versionSegment">The version from the <paramref name="fullVersion"/>, e.g. 1.2.3</param>
+    /// <param name="preReleaseSegment">The prerelease version from the <paramref name="fullVersion"/>, e.g. test</param>
+    /// <param name="buildSegment">The build version from the <paramref name="fullVersion"/>, e.g. node</param>
+    /// <returns></returns>
+    public static bool TryParseVersionComponents(
```

Comment:
not blocking: TryParseVersionComponents would benefit from unit tests (at least it would provide examples of expected input/output).  I think buildconfiggen doesn't currently have a unit test project - so this would take some churn to add and decide when to run the unit tests.  My opinion - run them in make.js using "dotnet test" prior to "dotnet run", especially if it just takes a few seconds

Due to additional chrun to add the infrastructure, I'd suggest deferring to a seperate PR